dazel.py

```diff
@@ -213,47 +241,71 @@ def start(self):
 
     def is_running(self):
         """Checks if the container is currently running."""
-        command = "%s ps | grep \"\\<%s\\>\" >/dev/null 2>&1" % (
-            self.docker_command, self.instance_name)
-        command = self._with_docker_machine(command)
+        command = self._with_docker_machine(
+            '{} ps  --no-trunc --filter name=^/{}$'.format(
+                self.docker_command,
+                self.instance_name))
+        output = self._run_command(command)
+        is_running = self._string_exists(self.instance_name, output)
 
         # If we have a directory, make sure the running container is mapped to
         # the same one (if not we need to create a new container mapped to the
         # correct folder).
-        if self.directory:
+        if self.directory and is_running:
             real_directory = os.path.realpath(self.directory)
-            command += (" && docker inspect \"%s\" | grep \"%s:%s\" >/dev/null 2>&1" %
-                        (self.instance_name, real_directory, real_directory))
+            dir_string = '{}:{}'.format(real_directory, self.remote_directory)
+            command = self._with_docker_machine(
+                '{} inspect --format="{}" "{}"'.format(
+                    self.docker_command,
+                    "{{json .HostConfig.Binds}}",
+                    self.instance_name))
+            output = self._run_command(command).strip()
+            binds = json.loads(output)
+            is_running = any(dir_string in b for b in binds)
 
         # If we have a network, make sure the running container is using the
         # correct network (if not we need to create a new container on the
         # correct network).
         # Note: with proper naming conventions this shouldn't happen much.
-        if self.network:
-            command += (" && docker inspect \"%s\" | grep '\"NetworkMode\": \"%s\"' >/dev/null 2>&1" %
-                        (self.instance_name, self.network))
+        if self.network and is_running:
+            command = self._with_docker_machine(
+                '{} inspect --format="{}" "{}"'.format(
+                    self.docker_command,
+                    "{{.HostConfig.NetworkMode}}",
+                    self.instance_name))
+            output = self._run_command(command).strip()
+            is_running = (output == self.network)
+        return is_running
+
+    def _run_silent_command(self, command, ignore_errors=False):
```

Comment:
Shouldn't this be `ignore_output` instead of `ignore_errors`?

Reply:
Done